test/iron-input.html

```diff
@@ -138,6 +163,27 @@
         assert.equal(input.bindValue, '123');
       });
 
+      test('prevent invalid input works in _onInput and does not use pattern', function() {
+        var input = fixture('prevent-invalid-input-with-pattern');
+        input.value = '123';
+        input._onInput();
+        assert.equal(input.bindValue, '123');
+        input.validate();
```

Comment:
This test is testing that `validate` works, not `prevent-invalid-input`. As I mentioned in the original issue, `validate` uses `pattern` regardless of what `prevent-invalid-input` does.

Maybe this test should be renamed to `inputs can be validated`? Which is still a valid test :)